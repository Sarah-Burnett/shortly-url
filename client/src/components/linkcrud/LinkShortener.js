import React from 'react';
import styled from 'styled-components';
import { breakpoint, violet } from '../../Styles';
import { LinkAdder } from './LinkAdder';
import imgDesktop from '../../images/bg-shorten-desktop.svg';
import imgMobile from '../../images/bg-shorten-mobile.svg';

const Shortener = styled.div`
    position: relative;
    width: 80vw;
    background-image: url(${imgMobile});
    background-color: ${violet};
    min-height: 30vh;
    margin: 2vh auto;
    border-radius: 15px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    @media (min-width: ${breakpoint}) {
        top: -10vh;
        background-image: url(${imgDesktop})
    }
`

export const LinkShortener = ({position}) => {
    return (
        <Shortener style={{top: position}}>
            <LinkAdder/>
        </Shortener>
    )
}
