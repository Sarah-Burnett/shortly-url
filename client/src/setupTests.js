import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, fireEvent, screen, waitForElement } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import renderer from 'react-test-renderer';
import { UserProvider } from './context/UserContext';
import { LinksProvider } from './context/LinksContext';
import { ActiveLinkProvider } from './context/ActiveLinkContext';
import { ProcessLinkProvider } from './context/ProcessLinkContext';
import { FlashProvider } from './context/FlashContext';
import App from './App';

export const ComponentWrapper = (Component) => {
    return (
        <UserProvider>
            <LinksProvider>
                <ActiveLinkProvider>
                    <ProcessLinkProvider>
                        <FlashProvider>
                            <Component/>
                        </FlashProvider>
                    </ProcessLinkProvider>
                </ActiveLinkProvider>
            </LinksProvider>
        </UserProvider>
    )
}

export const rendersWithoutCrashing = (Component) => {
    const div = document.createElement('div');
    ReactDOM.render(<Component/>, div);
}

export const matchesSnapshot = (Component) => {
    const tree = renderer.create(<Component/>).toJSON();
    expect(tree).toMatchSnapshot();
}

export const sampleLinks = [
    {
        "date":"2020-08-14T17:29:53.041Z",
        "click":[
            {"date":1597426302926},
            {"date":1597426306071},
            {"date":1597426310231},
            {"date":1597426314763},
            {"date":1597428206995},
            {"date":1597428210093},
            {"date":1597428215131},
            {"date":1597428218755}
        ],
        "_id":"K1pjMCK-F",
        "full":"http://reactcommunity.org/react-transition-group/css-transition",
        "userId":"5ebe7d235e4a501ee8699058",
        "short":"octopus","__v":8
    },
    {
        "date":"2020-08-14T17:33:55.748Z",
        "click":[],
        "_id":"lsncCe5RV",
        "full":"https://ba-grant.netlify.app/",
        "userId":"5ebe7d235e4a501ee8699058",
        "short":"bagrant","__v":0
    }
];
