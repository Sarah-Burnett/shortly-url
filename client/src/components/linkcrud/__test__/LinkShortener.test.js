import { rendersWithoutCrashing, matchesSnapshot, ComponentWrapper } from '../../../setupTests';
import LinkShortener from '../LinkShortener';

const LinkShortenerWrapper = () => ComponentWrapper(LinkShortener)

it('renders link shortener without crashing', () => rendersWithoutCrashing(LinkShortenerWrapper));

it('link shortener matches snapshot', () => matchesSnapshot(LinkShortenerWrapper));

//TODO: check css positioning of the form