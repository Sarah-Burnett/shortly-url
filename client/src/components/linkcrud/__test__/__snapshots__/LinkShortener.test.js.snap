// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link shortener matches snapshot 1`] = `
<div
  className="sc-fzqBZW deRmIF"
  style={
    Object {
      "top": undefined,
    }
  }
>
  <form
    className="sc-fznyAO dlGfaB"
    onSubmit={[Function]}
  >
    <label
      htmlFor="link"
      style={
        Object {
          "display": "none",
        }
      }
    >
      Website Url
    </label>
    <p
      className="sc-fzokOt kPqHuq"
    />
    <input
      className="sc-fznKkj jhFdRJ"
      data-testid="input"
      id="link"
      name="link"
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      value="Shorten a link here... "
    />
    <input
      className="sc-fznKkj sc-fznZeY iOAXEI"
      data-testid="button"
      disabled={false}
      type="submit"
      value="Shorten It!"
    />
  </form>
</div>
`;
